wardrobe/tradable_kitties/src/lib.rs

```diff
@@ -0,0 +1,446 @@
+                    return Err(TradeableKittyError::CannotBuyMoreThanOneKittyAtTime);
+                }
+            };
+        } else {
+            return Err(TradeableKittyError::BadlyTyped);
+        }
+    }
+
+    // Ensuring we found kitty in the input
+    ensure!(input_ktty_to_be_traded != None, {
+        TradeableKittyError::InputMissingError
+    });
+
+    // Seperate the coin and tdkitty in to seperate vecs from the output_data .
+    for utxo in output_data {
+        if let Ok(coin) = utxo.extract::<Coin<ID>>() {
+            let utxo_value = coin.0;
+            ensure!(
+                utxo_value > 0,
+                TradeableKittyError::MoneyError(MoneyError::ZeroValueCoin)
+            );
+            output_coin_data.push(utxo.clone());
+            // Process Coin
+        } else if let Ok(td_output_kitty) = utxo.extract::<TradableKittyData>() {
+            // Checking if more than 1 kitty in output is sent for trading.
+            match output_ktty_to_be_traded {
+                None => {
+                    // 1st tradable kitty is received in the output.
+                    output_ktty_to_be_traded = Some(td_output_kitty.clone());
+                    ensure!(
+                        input_ktty_to_be_traded.clone().unwrap().kitty_basic_data
+                            == td_output_kitty.kitty_basic_data, // basic kitty data is unaltered
+                        TradeableKittyError::KittyBasicPropertiesAltered // this need to be chan
+                    );
+                }
+                Some(_) => {
+                    // More than 1 tradable kitty are sent in output for trading.
+                    return Err(TradeableKittyError::CannotBuyMoreThanOneKittyAtTime);
+                }
+            };
+        } else {
+            return Err(TradeableKittyError::BadlyTyped);
+        }
+    }
+
+    // Ensuring we found kitty in the output
+    ensure!(output_ktty_to_be_traded != None, {
+        TradeableKittyError::InputMissingError
+    });
+
+    // Ensuring total money sent is enough to buy the kitty
+    ensure!(
+        total_price_of_kitty <= total_input_amount,
+        TradeableKittyError::InsufficientCollateralToBuyKitty
+    );
+
+    // Filterd coins sent to MoneyConstraintChecker for money validation.
+    MoneyConstraintChecker::<0>::Spend.check(&input_coin_data, &[], &output_coin_data)?;
+    Ok(())
+}
+
+/// checks if tradable kitty price updates is possible of not.
+/// Price of multiple tradable kitties can be updated in the same txn.
+fn check_kitty_price_update(
+    input_data: &[DynamicallyTypedData],
+    output_data: &[DynamicallyTypedData],
+) -> Result<TransactionPriority, TradeableKittyError> {
+    ensure!(
+        input_data.len() == output_data.len() && !input_data.is_empty(),
+        { TradeableKittyError::NumberOfInputOutputMismatch }
+    );
+
+    let mut dna_to_tdkitty_set: BTreeSet<KittyDNA> = BTreeSet::new(); // to check the uniqueness in input
```

Comment:
You don't need to do this. Tuxedo core already does it for you. Good fundamental thinking though!

https://github.com/Off-Narrative-Labs/Tuxedo/blob/1ad23430126711d6fca8899e4f00306e76bf98f8/tuxedo-core/src/executive.rs#L57-L65

Reply:
Removed the dna check.